File: Examples/Ruby/DocumentProperties/set_document_property.rb
#ExStart:
require 'aspose_slides_cloud'

class DocumentProperties

  include AsposeSlidesCloud
  include AsposeStorageCloud

  def initialize
    #Get App key and App SID from https://cloud.aspose.com
    AsposeApp.app_key_and_sid("", "")
    @slides_api = SlidesApi.new
  end

  def upload_file(file_name)
    @storage_api = StorageApi.new 
    response = @storage_api.put_create(file_name, File.open("../../../data/" << file_name,"r") { |io| io.read } )
  end

  # Set document property.
  def set_document_property
  	file_name = "sample-input.pptx"
  	upload_file(file_name)

  	property_name = "Author"
  	document_property = DocumentProperty.new
    document_property.name = property_name
    document_property.value = "Elon Musk"
  	response = @slides_api.put_slides_set_document_property(file_name, property_name, document_property)
  end

end

documentProperties = DocumentProperties.new()
puts documentProperties.set_document_property
#ExEnd:

File: Examples/Python/SetDocumentProperty.py
import asposeslidescloud
from asposeslidescloud.SlidesApi import SlidesApi
from asposeslidescloud.SlidesApi import ApiException
from asposeslidescloud.models import DocumentProperty

import asposestoragecloud
from asposestoragecloud.StorageApi import StorageApi
from asposestoragecloud.StorageApi import ResponseMessage

apiKey = "XXXXX" #sepcify App Key
appSid = "XXXXX" #sepcify App SID
apiServer = "http://api.aspose.com/v1.1"
data_folder = "../../data/"

#Instantiate Aspose Storage API SDK
storage_apiClient = asposestoragecloud.ApiClient.ApiClient(apiKey, appSid, True)
storageApi = StorageApi(storage_apiClient)
#Instantiate Aspose Slides API SDK
api_client = asposeslidescloud.ApiClient.ApiClient(apiKey, appSid, True)
slidesApi = SlidesApi(api_client);

#set input file name
name =  "sample-input.pptx"
propertyName = "Author"

body = DocumentProperty.DocumentProperty()
body.Name = "Author"
body.Value = "Farooq Sheikh"
body.BuiltIn = True

try:
    #upload file to aspose cloud storage
    response = storageApi.PutCreate(name, data_folder + name)

    #Aspose.Slides Cloud SDK API to set a single document property
    response = slidesApi.PutSlidesSetDocumentProperty(name, propertyName, body)

    if response.Status == "OK":
        docProperty = response.DocumentProperty
        print docProperty.Name + " :: " +  docProperty.Value

except ApiException as ex:
            print "ApiException:"
            print "Code:" + str(ex.code)
            print "Message:" + ex.message
